### DISCUSSION TITLE
[BUG] Set-PnPListPermission : The specified user could not be found

### DISCUSSION THREAD

**Discussion by YuriySamorodov** [pnp/powershell#4149] [General]
### Reporting an Issue or Missing Feature
I am reporting a bug

### Expected behavior 
The command `Set-PnPListPermission -Identity Documents -User user@outlook.com -AddRole 'Contribute'` should grant user contribute permission on **Documents** library.
### Actual behavior 
`Set-PnPListPermission -Identity Documents -User user@outlook.com -AddRole 'Contribute` returns an error:
`Set-PnPListPermission: The specified user user@outlook.com could not be found.`
![image](https://github.com/user-attachments/assets/f9caae6a-0c5b-43aa-9b3b-254fa29bb899)

This is also mentioned in #4009 

### Steps to reproduce behavior
```
Connect-PnPOnline -Url https://tenant.sharepoint.com/sites/sitename
Set-PnPListPermission -Identity Documents -User user@outlook.com -AddRole 'Contribute
```

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
PnP,PowerShell **2.5.39**

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@YuriySamorodov - I have tried with `2.5.42` from today and had no issues assigning the permission. Please try with the latest nightly or make sure your user is present.

**Comment by YuriySamorodov** [comment]
Thank you, @veronicageek . Just to confirm - have you tried to assign permission on the new user which did not exist in the particular SharePoint site?

**Comment by veronicageek** [comment]
Yes that's correct.

**Comment by jackpoz** [comment]
Did the user exist on the tenant or was it a guest not yet invited ?
You can run `New-PnPUser -LoginName user@outlook.com` to test if the user exists in the tenant or not.

**Reply by YuriySamorodov** [reply]
Nope, the user does not exist. I have also tried `New-PnPUser -LoginName user@outlook.com` but it failed with the same error `The specified user user@outlook.com could not be found`

**Reply by jackpoz** [reply]
Then you need to add the user to the tenant first or invite him.
